Formula/mariadb.rb

```diff
@@ -81,7 +72,6 @@ def install
       -DINSTALL_DOCDIR=share/doc/#{name}
       -DINSTALL_INFODIR=share/info
       -DINSTALL_MYSQLSHAREDIR=share/mysql
-      -DWITH_READLINE=yes
```

Comment:
Why would we remove this?

Reply:
Sorry, as I explained above, it would not build, therefore I removed it. It's not a critical component, and editline is used instead. Which is the default for the official Debian/Ubuntu builds.
```
$ mysql --version
mysql  Ver 15.1 Distrib 10.6.6-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper
```

Reply:
Can you try and put it back so we can see in CI.

Reply:
Done

Reply:
It's trying to use the bundled `readline` (bad) but then also uses the system `libedit` headers (worse). We can remove `WITH_READLINE`, but we also need to add
```ruby
on_linux do
  depends_on "readline" # uses libedit on macOS
end
```
to get this to build on Linux.

Reply:
I'm not sure why a dependency is needed when readline support would be disabled. But I could add this.

Reply:
I don't think you can disable it. It'll just use a bundled `readline` if you don't provide one.

Reply:
Thank you for your clarification. I made the changes.